javascript/ql/src/semmle/javascript/security/dataflow/ExceptionXss.qll

```diff
@@ -47,6 +63,50 @@ module ExceptionXss {
     NotYetThrown() { this = "NotYetThrown" }
   }
 
+  // Consider using "if (err) {.. [do something with err] .. }" as an extra condition if there are too many FP's.
+  class Callback extends DataFlow::FunctionNode {
+    Callback() {
+      exists(DataFlow::CallNode call | call.getLastArgument().getAFunctionValue() = this) and
+      this.getNumParameter() = 2 and
+      this.getParameter(0).getName().regexpMatch("err.*") // Using "e" was considered. But that matches too many jQuery methods where "element" is shortened as "e".
+    }
+
+    DataFlow::Node getErrorParam() { result = this.getParameter(0) }
+  }
+
+  DataFlow::CallNode getEnclosingCallNode(DataFlow::Node node) {
+    result.getEnclosingExpr() = getEnclosingCall(node.getEnclosingExpr())
+  }
+
+  InvokeExpr getEnclosingCall(Expr e) {
```

Comment:
This entire function seems prone to misoptimizations with its many uses of transitive closures (gut feeling). Can you add a bit of documentation for the cases that you want it to flag?

Reply:
I've removed `getEnclosingCall` from the query for now.  
I'll look at re-introducing it later.